src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.DecryptToken.cs

```diff
@@ -23,59 +24,49 @@ public partial class JsonWebTokenHandler : TokenHandler
-        internal TokenDecryptionResult DecryptToken(
+        internal Result<string, ExceptionDetail> DecryptToken(
             JsonWebToken jwtToken,
             ValidationParameters validationParameters,
             BaseConfiguration configuration,
             CallContext? callContext)
         {
             if (jwtToken == null)
-                return TokenDecryptionResult.NullParameterFailure(jwtToken, nameof(jwtToken));
+                return ExceptionDetail.NullParameter(
+                    nameof(jwtToken),
+                    new StackFrame(true));
 
             if (validationParameters == null)
-                return TokenDecryptionResult.NullParameterFailure(jwtToken, nameof(validationParameters));
+                return ExceptionDetail.NullParameter(
+                    nameof(validationParameters),
+                    new StackFrame(true));
 
             if (string.IsNullOrEmpty(jwtToken.Enc))
-                return new TokenDecryptionResult(
-                    jwtToken,
-                    ValidationFailureType.TokenDecryptionFailed,
-                    new ExceptionDetail(
-                        new MessageDetail(TokenLogMessages.IDX10612),
-                        ExceptionDetail.ExceptionType.SecurityToken,
-                        new System.Diagnostics.StackFrame()));
-
-            var keysOrExceptionDetail = GetContentEncryptionKeys(jwtToken, validationParameters, configuration, callContext);
-            if (keysOrExceptionDetail.Item2 != null) // ExceptionDetail returned
-                return new TokenDecryptionResult(
-                    jwtToken,
-                    ValidationFailureType.TokenDecryptionFailed,
-                    keysOrExceptionDetail.Item2);
-
-            var keys = keysOrExceptionDetail.Item1;
-            if (keys == null)
-                return new TokenDecryptionResult(
-                    jwtToken,
-                    ValidationFailureType.TokenDecryptionFailed,
-                    new ExceptionDetail(
-                        new MessageDetail(
-                            TokenLogMessages.IDX10609,
-                            LogHelper.MarkAsSecurityArtifact(jwtToken, JwtTokenUtilities.SafeLogJwtToken)),
-                        ExceptionDetail.ExceptionType.SecurityTokenDecryptionFailed,
-                        new System.Diagnostics.StackFrame()));
+                return new ExceptionDetail(
+                    new MessageDetail(TokenLogMessages.IDX10612),
+                    ExceptionType.SecurityToken,
+                    new StackFrame(true));
+
+            (IList<SecurityKey>? contentEncryptionKeys, ExceptionDetail? exceptionDetail) result =
+                GetContentEncryptionKeys(jwtToken, validationParameters, configuration, callContext);
+
+            if (result.exceptionDetail != null)
```

Comment:
can `result` be null?

Reply:
Result is a value type and can therefore not be null. I did add an override for the empty initialisation to prevent the creation of an invalid Result object through extension.